src/Build/Logging/FancyLogger/FancyLogger.cs

```diff
@@ -0,0 +1,220 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Security;
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Microsoft.Build.Logging;
+
+namespace Microsoft.Build.Logging.FancyLogger
+{
+    /// <summary>
+    /// Represents an identifiable line within the log
+    /// </summary>
+    public class LogLine
```

Comment:
Style guideline: create a new file for each class. We're not 100% consistent on this but it's generally easier to keep things that way. There should be a Visual Studio refactoring to do this for you (and name the file correctly!).